server/room-battle.js

```diff
@@ -974,7 +974,13 @@ class RoomBattle extends RoomGames.RoomGame {
 				avatar: user ? '' + user.avatar : '',
 				team: team,
 			};
-			this.stream.write(`>player ${slot} ` + JSON.stringify(options));
+			this.stream.write(`>player ${slot} ${JSON.stringify(options)}`);
+			if (this.rated) {
+				new Ladders.LadderStore(this.format).getRating(player.userid).then(response => {
```

Comment:
The rating should already be available from the `Ready` objects. Maybe forward those, instead of redoing them here.

Reply:
Especially since this is actively creating a new `LadderStore`, which is not the intended API use and I think might mess up the ladder database.